astropy/table/sorted_array.py

```diff
@@ -0,0 +1,286 @@
+import numpy as np
+
+class SortedArray(object):
+    '''
+    Implements a sorted array container using
+    a list of numpy arrays.
+
+    Parameters
+    ----------
+    lines : `Table`
```

Comment:
Is there a good reason you need to bundle the row index column with the values column(s)?  It seems like this creates a lot of special handling (e.g. `len(lines.colnames) - 1` or `lines.columns[:-1]`) that ends up making the code harder to read and maintain.  Can it just be split into a `data` attribute and a `row_index` attribute?  

Is there a reason that `data` cannot or should not be a `Table` instead of a list of columns?